src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -117,7 +117,15 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
 
     UCT_IB_MLX5DV_SET(create_qp_in, in, opcode, UCT_IB_MLX5_CMD_OP_CREATE_QP);
     qpc = UCT_IB_MLX5DV_ADDR_OF(create_qp_in, in, qpc);
-    UCT_IB_MLX5DV_SET(qpc, qpc, st, UCT_IB_MLX5_QPC_ST_RC);
+    if (attr->super.qp_type == UCT_IB_QPT_DCI) {
```

Comment:
Seems you also have to pass `dc==1` to ` uct_ib_mlx5_qpc_cs_res(attr->super.max_inl_cqe[UCT_IB_DIR_RX], 0)` on line 143 below. Maybe in this case you will not need set `attr.super.max_inl_cqe[UCT_IB_DIR_RX] = 0;` in `uct_dc_mlx5_iface_create_dci`?

Reply:
nope, `dc=1` actually means DCT, i.e. either 64 or 0. In DCI it should be 0.

Reply:
ok